cEP-0009.md

```diff
@@ -0,0 +1,181 @@
+coala-quickstart to provide better analysis. This cEP proposes a mechnaism to
+represent and extract information from project files and an interface to utilize the extracted information.
+
+## Proposed Approach
+
+### 1. Implementation of the “Info” class
+
+A subclass of `Info` class will represent a specific kind of information. For example, the class `IgnorePathInfo` may represent “ignored file paths”.
+
+```py
+class Info:
+    description = "Some information"
+    # for validation
+    value_type = "data-type of the information value, can also be an ``Enum``"
+                 "of possible choices."
+
+    def __init__(self,
+                 source,
+                 value: value_type):
+        self.source = source
+        self.value = value
+        self.name = self.__class__.__name__
+        # optional metadata
+        self.scope = "bear-specific" #other possible values may be "global"
+
+    def get_value(self):
+        return self.value
+```
+
+
+### 2. Implementation of “InfoExtractor” class
+Any derived class of the `InfoExtractor` abstract class consists of:
+- Methods for parsing the project file.
+- Methods to find information in the parsed file.
+- A collection of `Info` objects representing the information extracted from a particular kind of project file.
+- Files to parse and extract information from.
+
+```py
+class InfoExtractor:
+    # Kinds of files supported by the InfoExtractor
+    supported_files = [**]
+
+    # Meta-data
+    supported_information_kinds = []
+
+    def __init__(self,
+                 target_files,
+                 project_directory):
+        self.target_files = target_files
+        self.directory = directory
+        self.information = dict()
+
+    def parse_file(self, filename):
+        """
+        Parses the given file and returns the parsed file.
+        """
+        raise NotImplementedError
+
+    def find_information(self, fname, parsed_file):
+        """
+        Returns a collection of ``Info`` instances.
+        """
+        raise NotImplementedError
+
+    def _add_info(self, info:Info):
+        """
+        Method to organize and add the supplied information in self.information attribute.
+        """
+        pass
+
+    def extract_information(self):
+        """
+        Extracts the inforamtion, saves in the object and returns it.
+        """
+        filenames = retrieve_files(self.target_files)
+
+        for file in filenames:
+            pfile = self.parse_file(file)
+            file_info = find_information(cls, pfile)
+            if file_info:
+                self._add_info(file, file_info)
+
+        return self.information
+
+    def get_information(self):
+        """
+        Return extracted information (if any)
+        """
+        return self.information
+
+    def get_specific_information(self, info_name):
+        return self.information.get(info_name)
+
+    @classmethod
+    def retrieve_files(cls, file_globs, directory):
```

Comment:
do we need this function? I think we have something that's exactly like this in `coalib`